client/app/user/profile/profile.controller.js

```diff
@@ -9,23 +9,117 @@ angular.module('observatory3App')
+        //If you can access private data
+        if ($scope.isuser || $scope.loggedInUserRole === 'admin'){
+            $http.get('/api/users/' + $stateParams.id+'/private').success(function(user){
+                $scope.user = user;
+                if (!$scope.user.bio){
+                    $scope.user.bio = "An awesome RCOS developer!";
+                }
+                $scope.originalRole = user.role;
+                //Total attendance is current attendance in large and small groups
+                var u = $scope.user;
+                u.full={normal:0, bonus:0};
+                u.small={normal:0, bonus:0};
+                u.unverified = 0;
+                for (var b = 0; b < u.attendance.length; b++){
+                    var submission = u.attendance[b];
+                    if (!submission.verified){
+                        u.unverified +=1;
+                    }
+                    else if (submission.smallgroup && !submission.bonus){
+                        u.small.normal +=1;
+                    }
+                    else if (submission.smallgroup){
+                        u.small.bonus +=1;
+                    }
+                    else if(submission.bonus){
+                        u.full.bonus +=1;
+                    }
+                    else{
+                        u.full.normal +=1;
+                    }
+                }
+                u.full.all = user.totalDates.length;
+                u.small.all = user.totalSmallDates.length;
+
+                u.full.greyWidth = 0;
+                u.small.greyWidth = 0;
+
+                if (u.full.all === 0){
+                    u.full.goodWidth = 0;
+                    u.full.greyWidth = 100;
+                }
+                else{
+                    if (u.full.normal + u.full.bonus >= u.full.all){
+                        u.full.goodWidth = 100;
+                    }
+                    else{
+                        u.full.goodWidth = 100 * (u.full.normal + u.full.bonus) / u.full.all ;
+                    }
+                }
+
+                if (u.small.all === 0){
+                    u.small.goodWidth = 0;
+                    u.small.greyWidth = 100;
+                }
+                else{
+                    if (u.small.normal + u.small.bonus >= u.small.all){
+                        u.small.goodWidth = 100;
+                    }
+                    else{
+                        u.small.goodWidth = 100 * (u.small.normal + u.small.bonus) / u.small.all ;
+                    }
+                }
+
+                $http.get('/api/commits/user/' + user.githubProfile).success(function(commits){
+                    $scope.user.commits = commits;
+                    });
+                // get all users projects information
+                $scope.projects = [];
+                user.projects.forEach(function(projectId){
+                  $http.get("/api/projects/" + projectId).success(function(project){
```

Comment:
This was already in there and not in the scope of this pr. You should file an issue if it should be changed.